File: java/armstrong-numbers/src/main/java/ArmstrongNumbers.java
class ArmstrongNumbers {

	boolean isArmstrongNumber(int numberToCheck) {
		char[] numberAsString = ("" + numberToCheck).toCharArray();
		int resultingNumber = 0;

		for(char ch : numberAsString){
			resultingNumber += Math.pow( (double) Character.getNumericValue(ch), (double) numberAsString.length);
		}

		return resultingNumber == numberToCheck;
	}

}

File: csharp/armstrong-numbers/ArmstrongNumbers.cs
using System;
using System.Linq;

public static class ArmstrongNumbers
{
    public static bool IsArmstrongNumber(int number)
        => number.ToString()
        .Select(x => Math.Pow(char.GetNumericValue(x), number.ToString().Length))
        .Sum() == number;
}